File: sdk/cpp/src/api/topology.cpp
// Copyright 2026 Hypermesh Foundation. All rights reserved.
// Licensed under the Business Source License 1.1.

#include <hypermesh/api/topology.hpp>

namespace hypermesh {

TopologyInfo TopologyApi::info() const {
    auto j = http_.get("/api/v1/topology/info");
    return j.get<TopologyInfo>();
}

NeighborList TopologyApi::neighbors() const {
    auto j = http_.get("/api/v1/topology/neighbors");
    return j.get<NeighborList>();
}

} // namespace hypermesh

File: sdk/typescript/src/api/topology.ts
import type { HttpClient } from "../client.js";
import type { TopologyInfo, TopologyNeighbors } from "../types.js";

export class TopologyApi {
  constructor(private readonly http: HttpClient) {}

  async info(): Promise<TopologyInfo> {
    return this.http.get<TopologyInfo>("/api/v1/topology/info");
  }

  async neighbors(): Promise<TopologyNeighbors> {
    return this.http.get<TopologyNeighbors>("/api/v1/topology/neighbors");
  }
}
